paperqa/types.py

```diff
@@ -80,7 +93,43 @@ def matches_filter_criteria(self, filter_criteria: dict) -> bool:
 class Text(Embeddable):
     text: str
     name: str
-    doc: Doc
+    doc: DocDetails | Doc
+
+    @model_validator(mode="before")
+    @classmethod
+    def ensure_doc(cls, values: Any) -> Any:
+        if not isinstance(values, dict):
+            # NOTE: If I don't know what is being passed, I don't change it
+            # This happens in some tests
```

Comment:
Can you update this comment? Let's not leave mysterious comments in the code.

Try debugging here while running the unit tests and figuring out exactly what values it is